ls1intum/Artemis
# [Exam] Enable instructor code editor assignment repository

### Checklist
- [x] Server: Added integration tests

### Motivation and Context
The assignment repository allows instructors to easily check whether the tests for a programming exercises work as expected. This is especially helpful when setting up an exam.

### Description

- [x] Enable the assignment repository for the exam mode again
- [x] Close the repository properly before deleting it on 'Reset assignment repository'
- [x] Properly configure JGit repositories by disabling automatic garbage collection. Auto gc might lock files preventing a proper deletion. See https://stackoverflow.com/questions/45266021/java-jgit-files-delete-fails-to-delete-a-file-but-file-delete-succeeds and https://git-scm.com/docs/git-gc for a secription
- [x] As the assignment repository is just a student participation of the instructor, an additional argument was added to the `startExercise` method to identify this case. Otherwise, we don't have semantically correct information whether to lock the repository for this exam exercise or not.

### Steps for Testing


_Exam exercise_
1. Create a exam programming exercise
2. Open the instructor code editor
3. Create the assignment repository
4. Play around with the common code editor actions and verify that they work as expected
5. Hit the button "Delete assignment repository"
--> The assignment repository should be deleted
6. Re-create the assignment repository
7. Conduct the exam
--> No problems occur during preparation, participation and assessment

_Course exercise_
1. Create a course programming exercise
2. Participate in it
--> Everything should work like before



### Screenshots


![Capture](https://user-images.githubusercontent.com/16407766/90279206-0977cd00-de69-11ea-97ba-ec858e33022b.PNG)
Reset -> Delete
